Cargo.toml

```diff
@@ -31,6 +31,7 @@ thiserror = "1.0.61"
 deno_semver = { version = "0.5.8", optional = true }
 deno_package_json = { version = "0.1.2", optional = true }
 deno_path_util = "0.2.0"
+regex = "1.11.1"
```

Comment:
Let's improve this to not have a dependency on regex and it would maybe be better for knowledge of what a valid JSR package name is to not live in this crate. Maybe instead use `deno_semver::jsr::JsrPackageReqReference::from_str(&format!("jsr:{}@*", name))` ... and ensure the result doesn't have a `.sub_path()`?

Reply:
It might be nicer for us to add something to `deno_semver` for checking if a jsr package name is valid, but using `JsrPackageReqReference` might be ok.

Reply:
I used the hack to check validity. 

I had to remove the suggestion on expected package pattern to satisfy the statement. 
> it would maybe be better for knowledge of what a valid JSR package name is to not live in this crate